src/en/charms-offline-strategies.md

```diff
@@ -0,0 +1,145 @@
+machines can inter-communicate, all without going through a proxy.
+
+Hosts to exclude from the proxy include:
+
+ - localhost & 127.0.0.1 (standard 'no proxy' settings)
+ - the address assigned to the regular Ethernet device
+ - the address assigned to the host's LXD bridge (to talk to the controller via
+   its API)
+ - the entire subnet dedicated to the particular LXD installation
+
+## APT proxy
+
+An HTTP/S or FTP proxy may not be configured to accept requests for APT
+packages. An APT proxy is identical to that of an HTTP/S or FTP proxy except
+that it applies specifically to APT package requests.
+
+Some common implementations include:
+
+ - [`APT-cacher`][upstream-apt-cacher]
+ - [`Apt-Cacher NG`][upstream-apt-cacher-ng]
+ - [`squid`][upstream-squid]
+ - [`squid-deb-proxy`][upstream-squid-deb-proxy] (based on squid)
+
+Juju uses options `apt-ftp-proxy`, `apt-http-proxy`, and `apt-https-proxy` to
+set these proxies, depending on the protocol involved (i.e. FTP, HTTP, or
+HTTPS).
+
+## FTP proxy
+
+A standard FTP proxy. Juju uses the `ftp-proxy` option for this type of proxy.
+
+## APT mirror
+
+Instead of proxying client requests to an internet-based repository it is
+possible to maintain the repository internally. That is, you can have a copy or
+*mirror* of a Ubuntu package repository. This option has a large storage
+requirement and the initial setup/download time is considerable. Regular mirror
+synchronization will also be needed.
+
+Here are some popular mirroring solutions:
+
+ - [`apt-mirror`][upstream-apt-mirror]
+ - [`debmirror`][upstream-debmirror]
+ - [`aptly`][upstream-aptly]
+
+Juju uses the `apt-mirror` option for this.
+
+### HTTP/S server
+
+For any sort of mirror, an HTTP/S server (i.e. a web server) will be required
+to respond to the actual client requests. These are the most common ones:
+
+ - [`Apache`][upstream-apache]
+ - [`nginx`][upstream-nginx]
+ - [`lighttpd`][upstream-lighttpd]
+
+## Juju agent mirror
+
+An agent that gets installed onto a Juju machine is proxied through the
+controller. If the latter cannot satisfy the request via its cache it will
+download the agent from the [official agent site][upstream-agents], and then
+pass it on to the machine. It is possible, however, to set up an agent mirror
+so the remote site is not solicited by the controller. Download the latest few
+agents and configure one of the aforementioned web servers accordingly.
+Updates to the mirrored agents will be needed as new versions of Juju itself
+become available. Another approach is to have the client download agents and
+use the `juju sync-agent-binaries` command to push them to a model.
+
+## Cloud images mirror
+
+A mirror can be made of the [official cloud images][upstream-cloud-images].
+This will primarily be useful for a localhost cloud (LXD) but a local OpenStack
+installation can also make use of such a mirror if LXD containers are put on
+its instances.
```

Comment:
this doesn't really explain how you would set such a thing up

Reply:
It's not supposed to. This page outlines **strategies**. I don't think we should be getting into implementation details for non-Juju things. Again, I'm open for suggestions on how to tackle this but keep in mind that if we do it in one place we'll need to do it in all places.

Reply:
More explicitly, it doesn't really explain what you are mirroring. I know there is a link there, but it is a text link. I think if I am required to mirror a URL, it should give me the url. Also, is it necessary to mirror the entire thing? I suspect not.

Reply:
I will expose the URL. As for what to mirror, I can suggest "all currently supported releases".

Reply:
Done